src/async/pitfalls/blocking-executor.md

```diff
@@ -0,0 +1,60 @@
+# Blocking the executor
+
+Most async runtimes only allow IO tasks to run concurrently.
+This means that CPU blocking tasks will block the executor and prevent other tasks from being executed.
+An easy workaround is to use async equivalent methods where possible.
+
+```rust,editable,compile_fail
```

Comment:
This example looks too complicated to me. I suggest starting with a simple version, then adding steps in the speaker notes for building on top of it. You already have some steps in the speaker notes, perhaps move some of the code blocks here for reference to the speaker notes.